polaris-core/src/main/java/org/apache/polaris/core/config/PolarisConfigurationStore.java

```diff
@@ -99,7 +99,25 @@ public interface PolarisConfigurationStore {
    */
   default <T> @Nonnull T getConfiguration(
       @Nonnull RealmContext realmContext, PolarisConfiguration<T> config) {
-    T result = getConfiguration(realmContext, config.key, config.defaultValue);
+    T result = getConfiguration(realmContext, config.key);
+
+    if (result == null && config.hasLegacyKeys()) {
+      for (String legacyKey : config.legacyKeys) {
+        result = getConfiguration(realmContext, legacyKey);
+        if (result != null) {
+          LOGGER.warn(
+              "Legacy configuration key '{}' is in use. Please use '{}' instead.",
+              legacyKey,
+              config.key);
+          break;
```

Comment:
Not sure we want a break here
1. If multiple legacy keys really are in use, we should want to log for each of them
2. Barring the above, you could just early return rather than `break`

Reply:
I don't think we can use an early return directly because we need to try and cast the value to its defined type. We could duplicate the code here but I think this is cleaner.